src/ralph/assets/models/assets.py

```diff
@@ -263,6 +258,14 @@ def increment_hostname(cls, prefix, postfix=''):
         else:
             return obj
 
+    @classmethod
+    def get_next_free_hostname(cls, prefix, postfix, fill=5):
+        try:
+            last_hostname = cls.objects.get(prefix=prefix, postfix=postfix)
+        except cls.DoesNotExist:
+            last_hostname = cls(prefix=prefix, postfix=postfix)
```

Comment:
no - there is only creating object, but there is no saving if `DoesNotExist` is raised - is should only be created when new hostname is issued (not fetched/checked what will be the next hostname)